code/modules/paperwork/photocopier.dm

```diff
@@ -276,47 +176,323 @@
+		return
+	if(folder)
+		folder.forceMove(get_turf(src))
+		if(ishuman(usr))
+			usr.put_in_hands(folder)
+		to_chat(usr, "<span class='notice'>You take \the [folder] out of \the [src].</span>")
+		folder = null
+
+/**
+  * An internal proc for checking if a photocopier is able to copy an object
+  *
+  * It performs early checks/returns to see if the copier has any toner, if the copier is powered/working,
+  * if the copier is currently perfoming an action, or if we've hit the global copy limit. Used to inform
+  * the player in-game if they're using the photocopier incorrectly (no toner, no item inside, etc)
+  * Arguments:
+  * * scancopy - If TRUE, cancopy does not check for an item on/inside the copier to copy, used for copying stored files
+  */
+/obj/machinery/photocopier/proc/cancopy(scancopy = FALSE) //are we able to make a copy of a doc?
+	if(stat & (BROKEN|NOPOWER))
+		return
+	if(copying) //are we in the process of copying something already?
+		to_chat(usr, "<span class='warning'>[src] is busy, try again in a few seconds.</span>")
+		return
+	if(!scancopy && toner <= 0) //if we're not scanning lets check early that we actually have toner
+		visible_message("<span class='notice'>A yellow light on [src] flashes, indicating there's not enough toner for the operation.</span>")
+		return
+	if(max_copies_reached)
+		visible_message("<span class='warning'>The printer screen reads \"MAX COPIES REACHED, PHOTOCOPIER NETWORK OFFLINE: PLEASE CONTACT SYSTEM ADMINISTRATOR\".</span>")
+		return
+	if(total_copies >= MAX_COPIES_PRINTABLE)
+		visible_message("<span class='warning'>The printer screen reads \"MAX COPIES REACHED, PHOTOCOPIER NETWORK OFFLINE: PLEASE CONTACT SYSTEM ADMINISTRATOR\".</span>")
+		message_admins("Photocopier cap of [MAX_COPIES_PRINTABLE] paper copies reached, all photocopiers are now disabled.")
+		max_copies_reached = TRUE
+	if(!check_mob() && (!copyitem && !scancopy)) //is there anything in or ontop of the machine? If not, is this a scanned file?
+		visible_message("<span class='notice'>A red light on [src] flashes, indicating there's nothing in [src] to copy.</span>")
+		return
+	return TRUE
+
+/**
+  * Public proc for copying items
+  *
+  * Determines what item needs to be copied whether it's a mob's ass, paper, bundle, or photo and then calls the respective
+  * proc for it. Most toner var changing happens here so that the faxmachine child obj does not need to worry about toner
+  * Arguments:
+  * * obj/item/C - The item stored inside the photocopier to be copied (obj/paper, obj/photo, obj/paper_bundle)
+  * * scancopy - Indicates that obj/item/C is a stored file, we need to pass this on to cancopy() so it passes the check
+  */
+/obj/machinery/photocopier/proc/copy(obj/item/C, scancopy = FALSE)
+	if(!cancopy(scancopy))
+		return
+	copying = TRUE
+	playsound(loc, 'sound/goonstation/machines/printer_dotmatrix.ogg', 50, 1)
+	if(istype(C, /obj/item/paper))
+		for(var/i = copies; i > 0; i--)
```

Comment:
Forgot to mention this, but the reverse loops in this proc can be changed to the Byond format like so:
```dm
for(var/i in copies to 1 step -1)
```
Which will count down from `copies` to `1` and stop.

Reply:
added